dvc/remote/base.py

```diff
@@ -547,19 +548,56 @@ def download(
         if to_info.scheme != "local":
             raise NotImplementedError
 
-        logger.debug("Downloading '{}' to '{}'".format(from_info, to_info))
+        makedirs(to_info.parent, exist_ok=True, mode=dir_mode)
 
-        name = name or to_info.name
+        if self.isdir(from_info):
+            makedirs(to_info, exist_ok=True, mode=dir_mode)
+            file_to_infos = (
+                to_info / file_to_info.relative_to(from_info)
+                for file_to_info in self.walk_files(from_info)
+            )
 
+            with ThreadPoolExecutor(max_workers=self.JOBS) as executor:
+                file_from_info = list(self.walk_files(from_info))
+                with Tqdm(
+                    file_from_info,
+                    total=len(file_from_info),
+                    desc="Downloading directory",
+                ) as file_from_info:
+                    return sum(
+                        executor.map(
+                            partial(
+                                self.single_file_download,
+                                name=name,
+                                no_progress_bar=True,
+                                file_mode=file_mode,
+                                dir_mode=dir_mode,
+                            ),
+                            file_from_info,
+                            file_to_infos,
+                        )
+                    )
+        else:
+            self.single_file_download(
```

Comment:
And no need for "else" since you return earlier
```suggestion
            return self.single_file_download(
```